test/test_host.go

```diff
@@ -96,3 +96,105 @@ func (s *HostSuite) TestNetworkedPersistentJob(t *c.C) {
 
 	t.Assert(string(body), c.Equals, "echocococo\n")
 }
+
+func (s *HostSuite) TestVolumeCreation(t *c.C) {
+	h := s.anyHostClient(t)
+
+	vol, err := h.CreateVolume("default")
+	t.Assert(err, c.IsNil)
+	t.Assert(vol.ID, c.Not(c.Equals), "")
+}
+
+func (s *HostSuite) TestVolumeCreationFailsForNonexistantProvider(t *c.C) {
+	h := s.anyHostClient(t)
+
+	vol, err := h.CreateVolume("non-existent")
+	t.Assert(err, c.NotNil)
+	t.Assert(vol.ID, c.Equals, "")
+}
+
+func (s *HostSuite) TestVolumePersistence(t *c.C) {
+	// most of the volume tests (snapshotting, quotas, etc) are unit tests under their own package.
+	// these tests exist to cover the last mile where volumes are bind-mounted into containers.
+
+	cluster := s.clusterClient(t)
+
+	// create a volume!
+	vol, err := s.anyHostClient(t).CreateVolume("default")
+	t.Assert(err, c.IsNil)
+
+	// FIXME: yes, there's massive code duplication here.  boil it down later.
```

Comment:
Extracted the repeated test job creation components and the "api" call to the test shell into functions.  Much better now.